flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -274,27 +278,73 @@ class CodeGenerator(
       constantFlags: Option[Array[(Int, Boolean)]],
       outputArity: Int,
       needRetract: Boolean,
-      needMerge: Boolean)
+      needMerge: Boolean,
+      needReset: Boolean)
   : GeneratedAggregationsFunction = {
 
     // get unique function name
     val funcName = newName(name)
     // register UDAGGs
     val aggs = aggregates.map(a => generator.addReusableFunction(a))
     // get java types of accumulators
-    val accTypes = aggregates.map { a =>
-      a.getClass.getMethod("createAccumulator").getReturnType.getCanonicalName
+    val accTypeClasses = aggregates.map { a =>
+      val accType = TypeExtractor.createTypeInfo(a, classOf[AggregateFunction[_, _]], a.getClass, 1)
```

Comment:
why doesn't the original approach work anymore?
IMO, it is better to avoid the TypeExtractor if possible, esp. when trying to extract generic types.